libtfhe/core/arithmetic/torus_utils_big.h

```diff
@@ -58,20 +60,25 @@ class TorusUtils<BigTorus> {
      * @param Msize discrete space size
      * @return torus value
      */
-    static void modSwitchToTorus(TORUS *res, const uint64_t mu, const uint64_t Msize, const ZModuleParams<TORUS> *params, Allocator alloc) {
+    static void
+    modSwitchToTorus(TORUS *res, const uint64_t mu, const uint64_t Msize, const ZModuleParams<TORUS> *params,
+                     Allocator alloc) {
         tfhe_backend::modSwitchToTorus(res, mu, Msize, params, std::move(alloc));
     }
 
     /**
-     * @brief Return infinity norm between 2 torus elements
+     * @brief Return absolute distance between 2 torus elements
      *
      * @param t1 first torus element
      * @param t2 second torus element
      * @return double value of the infinity norm
      */
-    static double normInftyDist(const TORUS *t1, const TORUS *t2, const ZModuleParams<TORUS> *params) {
-        abort(); //TODO the code below won't do mod 1 properly (e.g. dist(0.01,0.99) is wrong): add a distance function to the torus
-        return abs(TorusUtils<TORUS>::to_double(t1, params)-TorusUtils<TORUS>::to_double(t2, params)); // quick and dirty :P
+    static double distance(const TORUS *t1, const TORUS *t2, const ZModuleParams<TORUS> *params, Allocator alloc) {
+        TORUS *t = alloc.newObject<TORUS>(params, &alloc);
+        tfhe_backend::sub(t, t1, t2, params);
+        double diff = std::abs(TorusUtils<TORUS>::to_double(t, params));
```

Comment:
does to_double always return the centered representation in [-0.5,0.5] ?

Reply:
Yes. Maybe we must ensure this somehow.